instrumentation/github.com/gin-gonic/gin/otelgin/gintrace.go

```diff
@@ -85,6 +85,17 @@ func Middleware(service string, opts ...Option) gin.HandlerFunc {
 
 		// pass the span through the request context
 		c.Request = c.Request.WithContext(ctx)
+		defer func() {
+			// as we have created new http.Request object we need to make sure that temporary files created to hold MultipartForm
+			// files are cleaned up. This is done by http.Server at the end of request lifecycle but Server does not
+			// have reference to our new Request instance therefore it is our responsibility to fix the mess we caused.
+			//
+			// This means that when we are on returning path from handler middlewares up in chain from this middleware
+			// can not access these temporary files anymore because we deleted them here.
+			if c.Request.MultipartForm != nil {
```

Comment:
I'm wondering, if there's another middleware layer outside this one, would it be removed because of this?

Reply:
It's hard to say, but in all cases, regardless of how many middlewares are chained, temporary files should be removed. I didn't check the return error of `RemoveAll` here, which aligns with the behavior in both Go's [`http` package](https://github.com/golang/go/blob/44a6f817ea0fbeb3ba4aa398794c4e80dba13b1e/src/net/http/server.go#L1727) and [echo-contrib](https://github.com/labstack/echo-contrib/pull/90). I believe each middleware should independently attempt to remove temporary files.

Reply:
see: https://github.com/open-telemetry/opentelemetry-go-contrib/pull/6609#issuecomment-2587572600